src/fondant/pipeline/compiler.py

```diff
@@ -515,3 +517,167 @@ def _set_configuration(self, task, fondant_component_operation):
+                f"{metadata.component_id}/manifest.json",
+            ],
+        )
+
+        # add arguments if any to command
+        for key, value in arguments.items():
+            if isinstance(value, (dict, list)):
+                command.extend([f"--{key}", f"'{json.dumps(value)}'"])
+            else:
+                command.extend([f"--{key}", f"'{value}'"])
+
+        # resolve dependencies
+        if dependencies:
+            for dependency in dependencies:
+                # there is only an input manifest if the component has dependencies
+                command.extend(
+                    [
+                        "--input_manifest_path",
+                        f"{metadata.base_path}/{metadata.pipeline_name}/{metadata.run_id}/"
+                        f"{dependency}/manifest.json",
+                    ],
+                )
+
+        return command
+
+    def compile(
+        self,
+        pipeline: Pipeline,
+        output_path: str,
+        instance_type: str = "ml.t3.medium",
+        role_arn: t.Optional[str] = None,
+    ) -> None:
+        """Compile a fondant pipeline to sagemaker pipeline spec and save it
+        to a specified output path.
+
+        Args:
+            pipeline: the pipeline to compile
+            output_path: the path where to save the Kubeflow pipeline spec.
+            instance_type: the instance type to use for the processing steps
+            (see: https://aws.amazon.com/ec2/instance-types/ for options).
+            role_arn: the role arn to use for the processing steps,
+            if none provided the `sagemaker.get_execution_role()` role will be used.
+        """
+        run_id = pipeline.get_run_id()
+        path = pipeline.base_path
+        pipeline.validate(run_id=run_id)
+
+        component_cache_key = None
+
+        steps: t.List[t.Any] = []
+
+        with tempfile.TemporaryDirectory(dir=os.getcwd()) as tmpdirname:
+            for component_name, component in pipeline._graph.items():
+                component_op = component["fondant_component_op"]
+                component_cache_key = component_op.get_component_cache_key(
+                    previous_component_cache=component_cache_key,
+                )
+
+                metadata = Metadata(
+                    pipeline_name=pipeline.name,
+                    run_id=run_id,
+                    base_path=path,
+                    component_id=component_name,
+                    cache_key=component_cache_key,
+                )
+
+                logger.info(f"Compiling service for {component_name}")
+
+                command = self.build_command(
+                    metadata,
+                    component_op.arguments,
+                    component["dependencies"],
+                )
+                depends_on = [steps[-1]] if component["dependencies"] else []
+
+                script = self.generate_component_script(
```

Comment:
Could you rename `script` to `script_path`. At first glance it seems like the function is returning the script string